docs/articles/documentation/test-api/test-code-structure.md

```diff
@@ -3,12 +3,148 @@ layout: docs
 
-## test-controller
\ No newline at end of file
+TestCafe tests must be organized into categories called *fixtures*.
+A JavaScript file with TestCafe tests can contain one or more fixtures.
+
+To declare a test fixture, use the `fixture` function.
+
+```text
+fixture( fixtureName )
+```
+
+```text
+fixture `fixtureName`
+```
+
+Parameter     | Type   | Description
+------------- | ------ | ------------------------
+`fixtureName` | String | The name of the fixture.
+
+[Tests](#tests) that constitute a fixture go after this declaration.
+
+### Specifying the Start Webpage
+
+You can optionally use the fixture declaration to specify a webpage at which all tests in a fixture start.
+To do this, use the `page` function.
+
+```text
+page( url )
+```
+
+Parameter | Type   | Description
+--------- | ------ | ------------------------------------------------
+`url`     | String | The URL of the webpage at which the tests start.
+
+```js
+fixture `My fixture`
+    .page('http://example.com');
+```
+
+If the start page is not specified, it defaults to `about:blank`.
+
+> You can also call the [Navigate action](actions/navigate.md) at the beginning of a test
+> to navigate to the start page.
+
+## Tests
+
+To introduce a test, call the `test` function and pass test code inside.
+
+```text
+test( testName, func(t) )
+```
+
+Parameter  | Type     | Description
+---------- | -------- | --------------------------------------------------------------------
+`testName` | String   | The test name. Must be unique within a fixture.
+`func`     | Function | An asynchronous function that contains test code.
+`t`        | Object   | The [test controller](#test-controller) used to access test run API.
+
+```js
+fixture `MyFixture`;
+
+test('Test1', async t => {
+    /* Test 1 Code */
+});
+
+test('Test2', async t => {
+    /* Test 2 Code */
+});
+```
+
+You are free to structure code within a test in any manner, as well as you can
+reference any modules or libraries you need.
+
+TestCafe provides a set of [actions](actions/index.md) used to manipulate the tested webpage.
+To check that the page state matches the expected one, use [assertions](assertions.md).
+
+### Test Controller
+
+A *test controller* is an object that actually holds methods of test API. That is why it is
+passed to each function that is expected to contain server-side test code (like [test](#tests),
+[beforeEach](#initialization-and-clean-up) or [afterEach](#initialization-and-clean-up)).
+
+Use the test controller to call [test actions](actions/index.md), handle [browser dialogs](handling-native-dialogs.md),
+use the [wait function](pausing-the-test.md) or [invoke client-side code](executing-code-in-the-browser/index.md).
+
+```js
+import { expect } from 'chai';
+
+test('My Test', async t => {
+    await t
+        .click('#send-button')
+        .handleAlert()
+        .typeText('#input', 'Peter Parker')
+        .wait(1000);
+
+    expect(await t.eval(() => getSomethingOnTheClient())).to.be.true;
```

Comment:
It's a little bit misleading, since `getSomethingOnTheClient` is not defined on client side and not passed as dependency. It's better use comment here, e.g.:
```js
expect(await t.eval(() => {/* get something on the client */})).to.be.true;
```